test/intersections.jl

```diff
@@ -150,6 +150,38 @@
     @inferred someornone(s1, s2)
   end
 
+  @testset "Rays" begin
+    # rays in 2D
```

Comment:
tests for 2D
![](https://user-images.githubusercontent.com/67096719/169515421-96968574-5b52-48a6-8cb0-b73cb4b6c9a8.png)
Please give feedback for 2D tests, then I will add tests for 3D

What else has to be tested?